src/app/shared/components/template/services/template-process.service.ts

```diff
@@ -0,0 +1,79 @@
+import { Injectable } from "@angular/core";
+import { ActivatedRoute, Router } from "@angular/router";
+import { FlowTypes } from "packages/plh-data/model";
+import { SettingsService } from "src/app/pages/settings/settings.service";
+import { TEMPLATE } from "src/app/shared/services/data/data.service";
+import { TourService } from "src/app/shared/services/tour/tour.service";
+import { TemplateContainerComponent } from "../template-container.component";
+import { TemplateNavService } from "./template-nav.service";
+import { TemplateTranslateService } from "./template-translate.service";
+import { TemplateVariablesService } from "./template-variables.service";
+import { TemplateService } from "./template.service";
+
+/**
+ * The template process service is a slightly hacky wrapper around the template container component so that
+ * we can process templates programatically (i.e. without rendering a template component)
+ *
+ * TODO - Ideally all core processing lobic should move into the template process service and the container
+ * component extending it (instead of vice-versa)
+ */
+@Injectable({ providedIn: "root" })
+export class TemplateProcessService {
+  container: TemplateContainerComponent;
+  constructor(
+    templateService: TemplateService,
+    templateVariables: TemplateVariablesService,
+    templateTranslateService: TemplateTranslateService,
+    tourService: TourService,
+    router: Router,
+    route: ActivatedRoute,
+    // elRef: ElementRef,
+    templateNavService: TemplateNavService,
+    // cdr: ChangeDetectorRef,
+    settingsService: SettingsService
+  ) {
+    // Create mock template container component
+    console.log("hello template process service");
+    this.container = new TemplateContainerComponent(
+      templateService,
+      templateVariables,
+      templateTranslateService,
+      tourService,
+      router,
+      route,
+      null as any,
+      templateNavService,
+      null as any,
+      settingsService
+    );
+  }
+
+  public async init() {
+    await this.initialiseStartupTemplates();
+  }
+
+  public async processTemplateWithoutRender(template: FlowTypes.Template) {
+    console.log("[Template Process]", template.flow_name);
+    // this.container.name = this.container.name || this.templatename;
+    // this.templateBreadcrumbs = [...(this.parent?.templateBreadcrumbs || []), this.name];
+    this.container.template = template;
+    // reset any existing templateRowMap data
+    this.container.templateRowService.templateRowMap = {};
+    // process the template as if it were rendered
+    // TODO - should filter out template rows to only include those used programatically (e.g. set_variable, set_field etc.)
+    await this.container.templateRowService.processContainerTemplateRows();
+  }
+
+  private async initialiseStartupTemplates() {
+    const startupTemplates = TEMPLATE.filter((t) => t.process_on_start).sort(
+      (a, b) => a.process_on_start - b.process_on_start
+    );
```

Comment:
This looks like ascending order I thought we agreed descending? 
The ordering doesn't actually matter for now but this would be good to review together later.